server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java

```diff
@@ -1854,6 +1854,22 @@ public void shouldSkipParamResoulutionForElasticConfig_asPartOf100To101Migration
         );
     }
 
+
+    @Test
+    public void shouldRemoveAgentWithDuplicateElasticAgentId_asPartOf102To103Migration() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+        String configXml = "<cruise schemaVersion='102'>" +
+                "<agents>\n" +
+                "    <agent hostname=\"hostname\" ipaddress=\"127.0.0.1\" uuid=\"c46a08a7-921c-4e77-b748-6128975a3e7d\" elasticAgentId=\"16649813-4cb3-4682-8702-8e202824dd73\" elasticPluginId=\"elastic-plugin-id\" />\n" +
```

Comment:
@GaneshSPatil --- Change the order of duplicate agents. Just to make sure that translation will remove duplicates in any order.